src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -108,22 +110,36 @@ public JestElasticsearchClient(String address) {
   }
 
   public JestElasticsearchClient(Map<String, String> props) {
+    this(props, new JestClientFactory());
+  }
+
+  // visible for testing
+  protected JestElasticsearchClient(Map<String, String> props, JestClientFactory factory) {
     try {
       ElasticsearchSinkConnectorConfig config = new ElasticsearchSinkConnectorConfig(props);
       final int connTimeout = config.getInt(
           ElasticsearchSinkConnectorConfig.CONNECTION_TIMEOUT_MS_CONFIG);
       final int readTimeout = config.getInt(
           ElasticsearchSinkConnectorConfig.READ_TIMEOUT_MS_CONFIG);
 
+      final String username = config.getString(
+          ElasticsearchSinkConnectorConfig.CONNECTION_USERNAME_CONFIG);
+      final String password = config.getString(
+          ElasticsearchSinkConnectorConfig.CONNECTION_PASSWORD_CONFIG);
+
       List<String> address =
           config.getList(ElasticsearchSinkConnectorConfig.CONNECTION_URL_CONFIG);
-      JestClientFactory factory = new JestClientFactory();
-      factory.setHttpClientConfig(new HttpClientConfig.Builder(address)
+      HttpClientConfig.Builder builder = new HttpClientConfig.Builder(address)
           .connTimeout(connTimeout)
           .readTimeout(readTimeout)
-          .multiThreaded(true)
-          .build()
-      );
+          .multiThreaded(true);
+      if (username != null && !username.isEmpty() && password != null && !password.isEmpty()) {
```

Comment:
If we are doing only Basic Auth should we be we using preemptive auth?

Reply:
Yes, it is using preemptive auth on line 138.